backend/src/error.rs

```diff
@@ -0,0 +1,64 @@
+#![allow(clippy::module_name_repetitions)]
+use std::fmt;
+
+use axum::response::{IntoResponse, Response};
+use error_stack::Context;
+use hyper::StatusCode;
+use thiserror::Error;
+
+/// Generic error that wraps `error_stack::Context`.
+/// Generally used for notifying the client that some error occurred on the server
+///
+/// For the internal server errors
+#[derive(Debug)]
+pub struct InternalAppError;
```

Comment:
I see 2 possible ways to work with errors:
1. For small projects it is ok to add `error.rs` file (like you do) and declare a single crate-wide error type (usually as enum) and use it everywhere
2. For large projects it is better to declare a separate error type in every module (again usually as enum) and use it only within that module

I think we can try the second way here, because `error_stack` authors created library for this primary scenario. In this case error types should be declared in every module, when they will be created. In this particular file you can declare a single top-level error only. Example:

```Rust
#[derive(Error, Debug)]
pub enum AppError {
    #[error(...)]
    InternalError,
    #[error(...)]
    InitializationError,  // This is example. The real error kinds should be added according to the real needs of the app
    ...
}
```

Reply:
In this comment, by "module" I mean an architectural element, not a Rust module (in rust every file is module and every directory is a module)